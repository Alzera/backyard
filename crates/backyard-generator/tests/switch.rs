use backyard_generator::generate;
use backyard_parser::parse_eval;

#[test]
fn basic() {
  let asts = parse_eval(
    "switch ($a) {
  case 1:
    break;
  case 2:
    return;
  default:
    continue;
}"
  ).unwrap();
  insta::assert_yaml_snapshot!(generate(&asts).unwrap());
}

#[test]
fn short() {
  let asts = parse_eval(
    "switch ($a):
  case 1:
    break;
  case 2:
    return;
  default:
    continue;
endswitch;"
  ).unwrap();
  insta::assert_yaml_snapshot!(generate(&asts).unwrap());
}

#[test]
fn case_bracket() {
  let asts = parse_eval("switch ($a) {
  case 1: {
    break;
  }
  default:
    continue;
}").unwrap();
  insta::assert_yaml_snapshot!(generate(&asts).unwrap());
}
