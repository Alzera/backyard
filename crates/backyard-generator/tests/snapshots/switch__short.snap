---
source: crates/backyard-generator/tests/switch.rs
expression: generate(asts).unwrap()
snapshot_kind: text
---
"switch ($a):\n  case 1:\n    break;\n  case 2:\n    return;\n  default:\n    continue;\nendswitch;"
